src/borrow_check/drop_check.md

```diff
@@ -0,0 +1,152 @@
+# Drop Check
+
+We generally require the type of locals to be well-formed whenever the
+local is used. This includes proving the where-bounds of the local and
+also requires all regions used by it to be live.
+
+The only exception to this is the implicitly dropping values when they
+go out of scope. This does not necessarily require the value to be live:
+
+```rust
+fn main() {
+    let x = vec![];
+    {
+        let y = String::from("I am temporary");
+        x.push(&y);
+    }
+    // `x` goes out of scope here, after the reference to `y`
+    // is invalidated. This means that while dropping `x` its type
+    // is not well-formed as it contain regions which are not live.
+}
+```
+
+This is only sound if dropping the value does not try to access any dead
+region. The code responsible for this is `dropck_outlives`.
```

Comment:
This is only sound if dropping the value does not try to access any dead
region. We check this by requiring the type of the value to be *drop-live*.
The requirements for which are computed in `fn dropck_outlives`.